src/cpufreq.c

```diff
@@ -28,18 +31,52 @@
 
 static int num_cpu = 0;
 
+static char const * freq_fname = "/sys/devices/system/cpu/cpu%d/cpufreq/"
+								 "scaling_cur_freq";
+static char const * freq_fname_rhel = "/sys/devices/system/cpu/cpu%d/cpufreq/"
+									  "cpuinfo_cur_freq";
+
+static int setup_freq_fname()
+{
+	int status;
+	char filename[256];
+
+	status = ssnprintf (filename, sizeof (filename),
+						freq_fname, 0);
+	if ((status < 1) || ((unsigned int)status >= sizeof (filename)))
+		return 0;
+
+	if (!access (filename, R_OK))
+		return 1;
+
+	status = ssnprintf (filename, sizeof (filename),
+						freq_fname_rhel, 0);
+	if ((status < 1) || ((unsigned int)status >= sizeof (filename)))
+		return 0;
+
+	if (!access (filename, R_OK)) {
+		freq_fname = freq_fname_rhel;
```

Comment:
Instead of overwriting `freq_fname`, I think it's cleaner to use three variables: the two you have now (the templates) and then another one to store the actual filename. That way, this function could safely be called multiple times. I'd rename the existing `freq_fname` variable to something else and then use the `freq_fname` name for the new variable.